server.go

```diff
@@ -2303,6 +2306,45 @@ func initNetworkBootstrappers(s *server) ([]discovery.NetworkPeerBootstrapper, e
 	return bootStrappers, nil
 }
 
+// createBootstrapIgnorePeers creates a map of peers that the bootstrap process
+// needs to ignore, which is made of three parts,
+//   - the node itself needs to be skipped as id doesn't make sense to connect
+//     to itself.
+//   - the peers that already have connections with, as in s.peersByPub.
+//   - the peers that we are attempting to connect, as in s.persistentPeers.
+func (s *server) createBootstrapIgnorePeers() map[autopilot.NodeID]struct{} {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	ignore := make(map[autopilot.NodeID]struct{})
```

Comment:
looks a bit weird to import something from the `autopilot` package for this. Can we not just use the pubkey string form like the `persistentPeers` and `peersByPub` maps do?

Reply:
ahh ok i see that `MultiSourceBootstrap` requires the ignore list to be of this type. So never mind :+1: